pkg/metrics/job_metrics.go

```diff
@@ -136,15 +136,28 @@ func (m *JobMetrics) RestartInc() {
 	m.restart.Inc()
 }
 
-func (m *JobMetrics) FirstPodLaunchDelaySeconds(job metav1.Object, status v1.JobStatus) {
-	cond := util.GetCondition(status, v1.JobRunning)
-	if cond == nil {
+func (m *JobMetrics) FirstPodLaunchDelaySeconds(activePods []*corev1.Pod, job metav1.Object, status v1.JobStatus) {
+	if !util.IsRunning(status) || status.StartTime == nil {
 		return
 	}
-	if status.StartTime == nil {
+
+	var earliestTime *metav1.Time
+	for _, pod := range activePods {
+		if pod.Status.Phase != corev1.PodRunning {
+			continue
+		}
+		readyCond := getPodCondition(&pod.Status, corev1.PodReady)
+		if readyCond == nil {
+			continue
+		}
+		if earliestTime == nil || readyCond.LastTransitionTime.Before(earliestTime) {
+			earliestTime = &readyCond.LastTransitionTime
+		}
+	}
+	if earliestTime == nil {
 		return
 	}
-	delay := metav1.Now().Time.Sub(status.StartTime.Time).Seconds()
+	delay := earliestTime.Time.Sub(status.StartTime.Time).Seconds()
```

Comment:
The StartTime for xgboost and xdl job currently are defined as ALL workers running. 
In that case, the earliestTime before the StartTime, may be we can use the job object's creationTimestamp

Reply:
done, creationTimestamp is a more appropriate choice.